internal/framework5provider/list_resource.go

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package framework
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/list"
+	listschema "github.com/hashicorp/terraform-plugin-framework/list/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/identityschema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var _ resource.Resource = ListResource{}
+var _ resource.ResourceWithIdentity = ListResource{}
+var _ resource.ResourceWithImportState = ListResource{}
+var _ list.ListResource = ListResource{}
+
+func NewListResource() resource.Resource {
```

Comment:
Just an observation, with the split in interfaces, maybe we should start recommending that these functions just return the structs themselves (especially since we already recommend/do interface assertions in the lines above (22-25))

Then you'd only need one:
```go
func NewListResource() *ListResource {
	return &ListResource{}
}
```

My guess is that this kind of method caught on to do the interface check in the same file that the struct was defined, but I'm actually not sure 🤔